src/fmu/dataio/_filedata_provider.py

```diff
@@ -175,7 +187,7 @@ def _get_path(self) -> tuple[Path, Path | None]:
 
     def _get_path_generic(
         self,
-        mode: Literal[FmuContext.REALIZATION, FmuContext.PREPROCESSED],
```

Comment:
@jcrivenaes Potential bug? I did not find any guards around the call-site that prohibits the other modes then the once listed above.